java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -158,49 +152,56 @@ class EntityHandleManager(
+        val storageAdapter: StorageAdapter<T, R>,
+        val particleId: String = ""
+    )
+
     private suspend fun <T : Storable, R : Referencable> createSingletonHandle(
-        handleName: String,
-        spec: HandleSpec<out Entity>,
-        storageKey: StorageKey,
-        storageAdapter: StorageAdapter<T, R>
+        config: HandleConfig<T, R>
     ): Handle {
-        val singletonHandle = SingletonHandle(
-            name = handleName,
-            spec = spec,
-            storageProxy = singletonStoreProxy(
-                storageKey,
-                spec.entitySpec.SCHEMA,
-                spec.dataType.toStorageMode()
+        val singletonConfig = SingletonHandle.Config(
+            name = config.handleName,
+            spec = config.spec,
+            proxy = singletonStoreProxy(
+                config.storageKey,
+                config.spec.entitySpec.SCHEMA,
+                config.spec.dataType.toStorageMode()
             ),
-            storageAdapter = storageAdapter,
-            dereferencerFactory = dereferencerFactory
+            storageAdapter = config.storageAdapter,
+            dereferencerFactory = dereferencerFactory,
+            particleId = config.particleId
         )
-        return when (spec.mode) {
+
+        val singletonHandle = SingletonHandle(singletonConfig)
+        return when (config.spec.mode) {
             HandleMode.Read -> object : ReadSingletonHandle<T> by singletonHandle {}
             HandleMode.Write -> object : WriteSingletonHandle<T> by singletonHandle {}
             HandleMode.ReadWrite -> object : ReadWriteSingletonHandle<T> by singletonHandle {}
-            else -> throw Error("Singleton Handles do not support mode ${spec.mode}")
+            else -> throw Error("Singleton Handles do not support mode ${config.spec.mode}")
         }
     }
 
     private suspend fun <T : Storable, R : Referencable> createCollectionHandle(
-        handleName: String,
-        spec: HandleSpec<out Entity>,
-        storageKey: StorageKey,
-        storageAdapter: StorageAdapter<T, R>
+        config: HandleConfig<T, R>
     ): Handle {
-        val collectionHandle = CollectionHandle(
-            name = handleName,
-            spec = spec,
-            storageProxy = collectionStoreProxy(
-                storageKey,
-                spec.entitySpec.SCHEMA,
-                spec.dataType.toStorageMode()
+        val collectionConfig = CollectionHandle.Config(
+            name = config.handleName,
+            spec = config.spec,
+            proxy = collectionStoreProxy(
+                config.storageKey,
+                config.spec.entitySpec.SCHEMA,
+                config.spec.dataType.toStorageMode()
             ),
-            storageAdapter = storageAdapter,
-            dereferencerFactory = dereferencerFactory
+            storageAdapter = config.storageAdapter,
+            dereferencerFactory = dereferencerFactory,
+            particleId = config.particleId
```

Comment:
To be honest, this doesn't seem like it's much of an improvement on what we had before. Particularly now that we need to copy across values between separate config objects... 

What are the benefits of the new config approach, in your view?

Reply:
Imo it'll make things easier to move to a builder pattern if we wanted.. also, you can re-use config objects even without a full builder.

Reply:
Thanks for explaining, sounds good.